### PR TITLE
added documentation index

### PR DESCRIPTION
**Pull request by WillemMali**
I added a README.md in the /docs directory which should help make the
docs more discoverable, and I added a reference to this index on the
main README.md. I also renamed 'tips.md' to 'kubectl.md' and changed
to title, to better reflect its content.

I also went through all the links on the page by hand to make sure they all
work.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2725)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

File: README.md [hunk 1]
```diff
@@ -56,8 +67,8 @@ $ chmod +x kops-linux-amd64                 # Add execution permissions
 $ mv kops-linux-amd64 /usr/local/bin/kops   # Move the kops to /usr/local/bin
 ```
 
-### Developer From Source
 
+### Developer From Source
```
Comment (chrislovecnm) [inline README.md]: Can we remove this section and move into developer docs?

File: README.md [hunk 2]
```diff
@@ -146,6 +161,7 @@ The maintainers and other community members are generally available on the [kube
  - Look at our [other interesting modes](/docs/commands.md#other-interesting-modes)
  - Full command line interface [documentation](/docs/cli/kops.md)
 
+
 ## GitHub Issues
 
 #### Bugs
```
Comment (chrislovecnm) [inline README.md]: Should be three pound signs

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
+Documentation Index
+===================
+
+This page serves as a categorized index for all `kops` documentation.
```
Comment (chrislovecnm) [inline docs/README.md]: I think we can remove this section. Kinda evident

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+Quick start
+-----------
+* [Getting started on AWS](aws.md)
+* [CLI reference](cli/kops.md)
+
+
+Overview
+--------
+
+* [CLI](#cli)
```
Comment (chrislovecnm) [inline docs/README.md]: Command line interface documentation

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+* [Introspection](#introspection)
```
Comment (chrislovecnm) [inline docs/README.md]: Not sure what u mean by this word.

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+* [Supported network topologies](topology.md)
+* [Subdomain setup](creating_subdomain.md)
+
+
+### Operations ###
+
+* [Cluster addon manager](addon_manager.md)
+* [Cluster addons](addons.md)
+* [Cluster configuration management](changing_configuration.md)
+* [Cluster with HA creation example](advanced_create.md)
```
Comment (chrislovecnm) [inline docs/README.md]: Spell out ha

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+* [Cluster upgrades and migrations](cluster_upgrades_and_migrations.md)
+* [`etcd` volume encryption setup](etcd_volume_encryption.md)
+* [`etcd` backup setup](etcd_backup.md)
+* [GPU setup](gpu.md)
+* [High Availability](high_availability.md)
```
Comment (chrislovecnm) [inline docs/README.md]: Not sure why we have two documents about ha

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+* [InstanceGroup images](images.md)
+    * how to modify IG images and information on available/tested images
```
Comment (chrislovecnm) [inline docs/README.md]: IG is specific kops lingo / jargon. What is another option to call it?

File: docs/README.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+* [Label management](labels.md)
+    * for AWS instance tags and `k8s` node labels
```
Comment (chrislovecnm) [inline docs/README.md]: Need to be more cloud generic with terminology. We have gcloud and vSphere support

File: docs/README.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+* [`kube-up` to `kops` migration](upgrade_from_kubeup.md)
+* [Secret management](secrets.md)
+* [Single-master to multi-master migration](single-to-multi-master.md)
+* [`k8s` updating](upgrade.md)
```
Comment (chrislovecnm) [inline docs/README.md]: Upgrade

Comment (chrislovecnm) [inline docs/README.md]: Upgrade not update

File: README.md (hunk 1 above)
Comment (WillemMali) [inline README.md]: It's already [in the developer docs](/kubernetes/kops/tree/master/docs/install.md), I'll remove the section altogether.

File: docs/README.md (hunk 5 above)
Comment (WillemMali) [inline docs/README.md]: Inspection perhaps? I mean: things you do when you look at the innards of a thing to debug it.

File: docs/README.md (hunk 8 above)
Comment (WillemMali) [inline docs/README.md]: Don't you think that if by the time a user wants to use custom IG images he knows what an IG is? It's also named in the "kops design documents" list, so a Ctrl+F would find it.

File: docs/README.md (hunk 9 above)
Comment (WillemMali) [inline docs/README.md]: Labels: remove the explanatory line? Call it "cluster node" labels?

IG: I went with cluster nodes again, it's a bit verbose but unambiguously indicates "full OS (like) images used on nodes in the cluster", while avoiding "Autoscaling group" which would personally make me dismiss the page as AWS-specific.

File: docs/README.md (hunk 10 above)
Comment (WillemMali) [inline docs/README.md]: Not very clear which lines you meant but I tried, see the new commit.

File: docs/README.md (hunk 7 above)
Comment (justinsb) [inline docs/README.md]: Probably because we never had an index!

**Comment by justinsb** [comment]
This is great - thank you @WillemMali

**Pull request merged**